sample/samplers.go

```diff
@@ -112,9 +132,12 @@ func NewSampler(temperature float32, topK int, topP float32, minP float32, seed
 		transforms = append(transforms, MinP(minP))
 	}
 
-	if seed >= 0 {
-		seed64 := uint64(seed)
-		return Weighted(&seed64, transforms...), nil
+	var rng *rand.Rand
+	if seed != -1 {
+		seed1 := uint64(seed)
+		seed2 := uint64(seed) ^ 0x12345678 // XOR with a constant to get a different but deterministic value
+		rng = rand.New(rand.NewPCG(seed1, seed2))
```

Comment:
```suggestion
		rng = rand.New(rand.NewPCG(uint64(seed), uint64(seed) + 1))
```

Is there a difference between adding 1 and `^ 0x12345678` ?

Reply:
Or maybe 

```
rand.New(rand.NewPCG(uint64(seed), 0x1))
```

I would aim for the logits that are produced with the same seed today in llama.cpp